File: JavaFundamentals/src/Arrays/Exercise/EqualSums.java
package Arrays.Exercise;

import java.util.Arrays;
import java.util.Scanner;

public class EqualSums {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int[] numbers = Arrays.stream(scanner.nextLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        boolean found = false;

        for (int i = 0; i < numbers.length; i++) {
            int leftSum = 0;
            int rightSum = 0;

            for (int left = 0; left < i; left++) {
                leftSum += numbers[left];
            }

            for (int right = i + 1; right < numbers.length; right++) {
                rightSum += numbers[right];
            }

            if (leftSum == rightSum) {
                System.out.println(i);
                found = true;
                break;
            }
        }

        if (!found) {
            System.out.println("no");
        }
    }
}

File: JS Fundamentals/Arrays - Exercise/equal_sums.js
function equalSums(arr) {
    let hasEqualSum = false;
    for(let index = 0; index < arr.length; index++){
        let leftSum = 0;
        let rightSum = 0;
        for(let ind = index - 1; ind >= 0; ind --){
            leftSum += arr[ind];
        }
        for(let ind = index + 1; ind < arr.length; ind++){
            rightSum += arr[ind];
        }
        if(leftSum === rightSum){
            console.log(index);
            hasEqualSum = true;
            break;
        }
    }
    if(!hasEqualSum){
        console.log('no')
    }
}

equalSums([1, 2, 3, 3]);
equalSums([1, 2]);
equalSums([1]);
equalSums([1, 2, 3]);
equalSums([10, 5, 5, 99, 3, 4, 2, 5, 1, 1, 4]);